### PR TITLE
docs: add more explanations on how to integrate aligned

### LINKED ISSUE #576: docs: rename `Using Aligned on your app` section to `Integrating Aligned into Your Application`

### PR DESCRIPTION
**Pull request by JuArce**
~~Merge after #554~~

### CODE REVIEW

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -27,7 +27,7 @@
 
 * [Submitting proofs](guides/0_submitting_proofs.md)
 * [SDK](guides/1_SDK.md)
-* [Using Aligned on your app](guides/2_using_aligned_on_your_app.md)
+* [Integration Aligned into your Application](guides/2_integration_aligned_into_your_application)
```
Comment (taturosati) [inline docs/SUMMARY.md]:
```suggestion
* [Integrating Aligned into your Application](guides/2_integration_aligned_into_your_application)
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+```
+
+### Step 4 - Verify the proof was verified in Aligned
+
+Once the proof is verified in Aligned, 
+you can verify that it was verified from your smart contract.
+
+The full example of this flow can be found on the [ZKQuiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
+
+An example [python script](../../examples/zkquiz/encode_verification_data.py) can be found to encode the call data from the json output of the Aligned cli.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
An example [python script](../../examples/zkquiz/encode_verification_data.py) can be found to encode the call data from the json output of the Aligned CLI.
```

Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md:145]: Remove space.

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+For SP1 this means having the [script](../../examples/zkquiz/quiz/script/src/main.rs)
+generate the proof.
+
+Then, submit the proof to Aligned for verification. 
+This can be done either with the SDK or by using the Aligned CLI.
+You can find examples of how to submit a proof using the cli 
+in the [submitting proofs guide](0_submitting_proofs.md).
+
+The call ZK Quiz uses is:
+```bash
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion

```bash
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        // Your function code ...
+    }
+}
+```
+
+You can find the example of the smart contract that checks the proof was verified in Aligned
+in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
+
+Note that the contract checks that the verification key commitment is the same as the program elf.
+```solidity
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion

```solidity
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+You can find an example of the full flow of using Aligned on your app in the [ZKQuiz example](../../examples/zkquiz). 
+
+## Steps
+
+### Step 1 - Generate your ZK Proof
+
+Generate your ZK proofs using any of the proving systems supported by Aligned.
+For this example, we use the SP1 proving system. The current SP1 version used in Aligned is v1.0.8-testnet.
+
+You can find the example of the quiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example](../../examples/zkquiz) directory.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
You can find an example of the quiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example](../../examples/zkquiz) directory.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+### Step 2 - Write your smart contract
+
+To check if a proof was verified in Aligned, you need to call to the Aligned ServiceManager contract inside your smart contract. 
+
+Also, you will need a way to check that the proven program is your own.
+
+The aligned cli provides a way for you to get the verification key commitment
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
The aligned CLI provides a way for you to get the verification key commitment
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        
+        // Your function code ...
+    }
+}
+```
+
+You can find the example of the smart contract that checks the proof was verified in Aligned
+in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
+
+Note that the contract checks that the verification key commitment is the same as the program elf.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
Note that the contract checks that the verification key commitment is the same as the program ELF.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+```solidity
+require(elfCommitment == provingSystemAuxDataCommitment, "ELF does not match");
+```
+
+This contract also includes a static call to the Aligned ServiceManager contract 
+to check that the proof was verified in Aligned.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
to check if the proof was verified in Aligned.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+require(callWasSuccessfull, "static_call failed");
+
+bool proofIsIncludedBool = abi.decode(proofIsIncluded, (bool));
+require(proofIsIncludedBool, "proof not included in batch");
+```
+
+### Step 3 - Have your app generate the proof and submit it to Aligned
+
+First, generate the proof. 
+For SP1 this means having the [script](../../examples/zkquiz/quiz/script/src/main.rs)
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
For SP1, this means having the [script](../../examples/zkquiz/quiz/script/src/main.rs)
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+generate the proof.
+
+Then, submit the proof to Aligned for verification. 
+This can be done either with the SDK or by using the Aligned CLI.
+You can find examples of how to submit a proof using the cli
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
You can find examples of how to submit a proof using the CLI 
```

File: docs/guides/2_integration_aligned_into_your_application.md (hunk 2 above)
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
An example [python script](../../examples/zkquiz/encode_verification_data.py) can be found to encode the call data from the JSON output of the Aligned CLI. 
```

**Review by glpecile — CHANGES_REQUESTED** [review]

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 11]
```diff
@@ -122,7 +121,83 @@ Then, submit the proof to Aligned for verification. This can be done either with
...
+    match wait_for_proof_verification(aligned_verification_data, rpc_url).await {
+        Ok(_) => println!("Proof verified"),
+        Err(err) => println!("Error: {:?}", err),
+    }
+}
+```
+
+You can find an example of the proof submission and verification in the [Quiz Program](../../examples/zkquiz/quiz/script/src/main.rs).
+
+The example generates a proof, instantiate a wallet to submit the proof, and then submits the proof to Aligned for verification. It then waits for the proof to be verified in Aligned.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
The example generates a proof, instantiates a wallet to submit the proof, and then submits the proof to Aligned for verification. It then waits for the proof to be verified in Aligned.
```

**Review by taturosati — APPROVED** [review]

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 12]
```diff
@@ -0,0 +1,238 @@
...
+
+You can find an example of the quiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example](../../examples/zkquiz) directory.
+
+### Step 2 - Write your smart contract
+
+To check if a proof was verified in Aligned, you need to call to the Aligned ServiceManager contract inside your smart contract.
+
+Also, you will need a way to check that the proven program is your own.
+
+The aligned CLI provides a way for you to get the verification key commitment without actually generating and submitting a proof.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
The Aligned CLI provides a way for you to get the verification key commitment without actually generating and submitting a proof.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 13]
```diff
@@ -0,0 +1,238 @@
...
+        
+        bool proofIsIncludedBool = abi.decode(proofIsIncluded, (bool));
+        require(proofIsIncludedBool, "proof not included in batch");
+        
+        // Your function code ...
+    }
+}
+```
+
+You can find the example of the smart contract that checks the proof was verified in Aligned in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
You can find an example of the smart contract that checks if the proof was verified in Aligned in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 14]
```diff
@@ -0,0 +1,238 @@
...
+
+Note that the contract checks that the verification key commitment is the same as the program elf.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
Note that the contract checks that the verification key commitment is the same as the program ELF.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 15]
```diff
@@ -0,0 +1,238 @@
...
+require(proofIsIncludedBool, "proof not included in batch");
+```
+
+### Step 3 - Submit and verify the proof to Aligned
+
+The proof submission and verification can be done either with the SDK or by using the Aligned CLI.
+
+#### Using the SDK
+
+To submit a proof using the SDK, you can use the `submit` function, and then you can use the `verify_proof_onchain` to check if the proof was correctly verified in Aligned.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
To submit a proof using the SDK, you can use the `submit` function, and then you can use the `verify_proof_onchain` function to check if the proof was correctly verified in Aligned.
```

**Review by glpecile — CHANGES_REQUESTED** [review]

**Review by glpecile — APPROVED** [review]
LGTM 👍

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 16]
```diff
@@ -0,0 +1,236 @@
+# Integrating Aligned into your Application
+
+Aligned can be integrated into your applications in a few simple steps to provide a way to verify ZK proofs generated inside your system.
+
+You can find an example of the full flow of using Aligned in your app in the [ZKQuiz example](../../examples/zkquiz).
+
+This example shows a sample app that generates an SP1 proof that a user knows the answers to a quiz, then submits the proof to Aligned for verification. Finally, it includes a smart contract that verifies that a proof was verified in Aligned and mints an NFT.
```
Comment (entropidelic) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
This example shows a sample app that generates an SP1 proof that a user knows the answers to a quiz and then submits the proof to Aligned for verification. Finally, it includes a smart contract that verifies that a proof was verified in Aligned and mints an NFT.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 17]
```diff
@@ -0,0 +1,236 @@
...
+        Err(err) => println!("Error: {}", err),
+    }
+}
+```
+
+You can find an example of the quiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example](../../examples/zkquiz) directory.
+
+## 2. Write your smart contract
+
+To check if a proof was verified in Aligned, you need to call to the Aligned ServiceManager contract inside your smart contract.
```
Comment (entropidelic) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
To check if a proof was verified in Aligned, you need to make a call to the `AlignedServiceManager` contract inside your smart contract.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+
+Also, you will need a way to check that the proven program is your own.
```
Comment (entropidelic) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
Also, you will need a way to check that the proven program is the correct one.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 19]
```diff
@@ -0,0 +1,236 @@
...
+        require(proofIsIncludedBool, "proof not included in batch");
+        
+        // Your function code ...
+    }
+}
+```
+
+You can find an example of the smart contract that checks if the proof was verified in Aligned in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
+
+Note that the contract checks that the verification key commitment is the same as the program ELF.
```
Comment (entropidelic) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
Note that the contract checks that the verification key commitment is the same as the program ELF commitment.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 20]
```diff
@@ -0,0 +1,236 @@
...
+    // Example user answers
+    let user_answers = "abc";
+    match generate_sp1_proof(user_answers) {
+        Ok(proof) => println!("Proof: {:?}", proof),
+        Err(err) => println!("Error: {}", err),
+    }
+}
+```
+
+You can find an example of the quiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example](../../examples/zkquiz) directory.
```
Comment (entropidelic) [inline docs/guides/2_integration_aligned_into_your_application.md]: This section is a little confusing for someone with no context. The rust code seems to be like a shorter version of the program to generate the proof in `../../examples/zkquiz/quiz/script/src/main.rs`. If we add this code we should add some more context (like why are we showing this code in the first place?) and exaplanation of what is happening

Comment (JuArce) [reply docs/guides/2_integration_aligned_into_your_application.md]: I agree. I will add an explanation on every code section

File: docs/SUMMARY.md [hunk 21]
```diff
@@ -24,7 +24,7 @@
 
 * [Submitting proofs](guides/0_submitting_proofs.md)
 * [SDK](guides/1_SDK.md)
-* [Using Aligned on your app](guides/2_using_aligned_on_your_app.md)
+* [Integrating Aligned into your Application](guides/2_integration_aligned_into_your_application.md)
```
Comment (glpecile) [inline docs/SUMMARY.md]:
```suggestion
* [Integrating Aligned into your Application](guides/2_integrating_aligned_into_your_application.md)
```

File: docs/guides/2_integration_aligned_into_your_application.md
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md:1]: Change name to `2_integrating_aligned_into_your_application.md`

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 22]
```diff
@@ -0,0 +1,201 @@
...
+
+You can find an example of the full flow of using Aligned in your app in the [ZKQuiz example](../../examples/zkquiz).
+
+This example shows a sample app that generates an SP1 proof that a user knows the answers to a quiz and then submits the proof to Aligned for verification. Finally, it includes a smart contract that verifies that a proof was verified in Aligned and mints an NFT.
+
+## 1. Generate your ZK Proof
+
+To submit proofs to Aligned and get them verified, first you need to generate those proofs. Every proving system has its own way of generating proofs.
+
+You can find examples of how to generate proofs in the [generating proofs guide](3_generating_proofs.md).
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
You can find examples on how to generate proofs in the [generating proofs guide](3_generating_proofs.md).
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+
+Also you can find an example of the ZKQuiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example](../../examples/zkquiz) directory.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
Also you can find an example of the ZKQuiz proof [program](../../examples/zkquiz/quiz/program/src/main.rs) as well as the [script](../../examples/zkquiz/quiz/script/src/main.rs) that generates it in the [ZKQuiz example directory](../../examples/zkquiz).
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 24]
```diff
@@ -0,0 +1,201 @@
...
+
+The Aligned CLI provides a way for you to get the verification key commitment without actually generating and submitting a proof.
+
+You can do this by running the following command:
+
+```bash
+aligned get-vk-commitment --input <path_to_input_file>
+```
+
+The following is an example of how to call the `verifyBatchInclusionMethod` from the Aligned ServiceManager contract in your smart contract.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
The following is an example of how to call the `verifyBatchInclusionMethod` from the `AlignedServiceManager` contract in your smart contract.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+        require(proofIsIncludedBool, "proof not included in batch");
+        
+        // Your function code ...
+    }
+}
+```
+
+You can find an example of the smart contract that checks if the proof was verified in Aligned in the [Quiz Verifier Contract](../../examples/zkquiz/contracts/src/VerifierContract.sol).
+
+Note that the contract checks that the verification key commitment is the same as the program ELF commitment.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
Note that the contract checks if the verification key commitment is the same as the program ELF commitment.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 26]
```diff
@@ -0,0 +1,201 @@
...
+
+```solidity
+require(elfCommitment == provingSystemAuxDataCommitment, "ELF does not match");
+```
+
+This contract also includes a static call to the Aligned ServiceManager contract to check if the proof was verified in Aligned.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
This contract also includes a static call to the `AlignedServiceManager` contract to check if the proof was verified in Aligned.
```

File: docs/guides/2_integration_aligned_into_your_application.md [hunk 27]
```diff
@@ -0,0 +1,201 @@
...
+    match wait_for_proof_verification(aligned_verification_data, rpc_url).await {
+        Ok(_) => println!("Proof verified"),
+        Err(err) => println!("Error: {:?}", err),
+    }
+}
+```
+
+You can find an example of the proof submission and verification in the [ZKQuiz Program](../../examples/zkquiz/quiz/script/src/main.rs).
+
+The example generates a proof, instantiates a wallet to submit the proof, and then submits the proof to Aligned for verification. It then waits for the proof to be verified in Aligned.
```
Comment (glpecile) [inline docs/guides/2_integration_aligned_into_your_application.md]:
```suggestion
This example generates a proof, instantiates a wallet to submit the proof, and then submits the proof to Aligned for verification. Lastly, it waits for the proof to be verified in Aligned.
```

**Review by glpecile — CHANGES_REQUESTED** [review]

**Review by glpecile — APPROVED** [review]

**Pull request merged**